portal/static/js/main.js

```diff
@@ -86,15 +86,66 @@ var ConsentUIHelper = function(consentItems, userId) {
      * relevant variables currently defined in profile_macro.html, but can be defined by consumer
      *
      */
-    this.ctop = (typeof CONSENT_WITH_TOP_LEVEL_ORG != "undefined") && CONSENT_WITH_TOP_LEVEL_ORG;
-    this.isAdmin = typeof isAdmin != "undefined" && isAdmin ? true: false;
-    this.editable = (typeof consentEditable != "undefined" && consentEditable == true) ? true : false;
-    this.consentDateEditable = this.editable && (typeof isTestPatient != "undefined" && isTestPatient);
+    this.ctop = false;
+    this.isAdmin = false;
+    this.editable = String($("#profileConsentEditable").val()).toLowerCase() === "true" ? true : false;
+    this.consentDateEditable = this.editable && (String($("#profileIsTestPatient").val()).toLowerCase() === "true");
     this.touObj = [];
     this.showInitialConsentTerms = false;
     this.hasConsent = false;
     this.hasHistory = false;
     this.orgTool = null;
+    this.initChecks = [];
+    this.userRoles = [];
+
+
+    const CONSENT_WITH_TOP_LEVEL_ORG = "CONSENT_WITH_TOP_LEVEL_ORG";
+
+
+    this.init = function(callback) {
+        var self = this;
+        self.initStartTime = new Date();
+
+        self.getOrgTool();
+        
+        /*
+         * check whether user is admin, can also check for other roles if need be
+         */
+        tnthAjax.getRoles($("#profileCurrentUserId").val(), false, function(data) {
```

Comment:
Would you be able to cache the roles in localStorage (and only make the request if necessary)?